### ISSUE TITLE
Layered Navigation with configurable product

### ISSUE THREAD

**Issue by mbattistini** [OpenMage/magento-lts#2663]
### Preconditions (*)

1. OpenMage 19.4.18
2. PHP 7.4

### Steps to reproduce (*)

1. Insert this products:

![products](https://user-images.githubusercontent.com/29307582/27280140-2dc3d60e-54e7-11e7-8212-f626fdab0097.png)

### Expected result (*)

1. Only configurable product A is visible

### Actual result (*)

1. Both configurable product A and B are visible. So if the customer clicks on the product B does not find the desired combination.

**Comment by ADDISON74** [comment]
I have not checked this report, but I see that it is confirmed in Magento 2 too https://github.com/magento/magento2/issues/36364

**Comment by mbattistini** [comment]
Yes, I also have sites with Magento 2 and the problem there is also there.
The Issue had already been open but I noticed that it is still present, I think it is also linked to the stock of the products, at least it is for Magento 2.x

**Comment by fballiano** [comment]
@mbattistini if I try to replicate on https://demo.openmage.org/men/pants-denim.html?apparel_type=36&fit=56 I select 2 filters but they seem to correctly work in "AND", in fact the result is a subset of the original pool of products.

are your length and width 2 different attributes?

**Comment by mbattistini** [comment]
you have to try to select attributes for which products are out of stock
and you will see that they appear even if they shouldn't.
We currently have up to 5 attributes on some products and the filter
system doesn't take into account whether THE SON (simple product) is in
stock or not.
Apparently the verification only happens on the FATHER (configurable
product).
Looks like they went a bit further with the analysis on magento 2, same
bug.
